DOCUMENTATION.md

```diff
@@ -43,6 +45,153 @@ The overall territory is laid out here in this [C4](https://c4model.com) System
+
+At the other end of the redelivery spectrum, we have full replays. For instance, it's not uncommon to want to either re-traverse an entire set of events (e.g. if some field was not being stored in the derived data but suddenly becomes relevant), or one may opt to rewind to an earlier checkpoint to trigger re-processing (a downstream processor reports having restored a two hour old back resulting in data loss, which could be resolved by rewinding 3 hours and relying on the idempotency guarantees of their APIs).
+
+A related scenario that often presents itself after a system has been running for some time is the desire to add an entirely new (or significantly revised) read model. In such as case, being able to traverse a large number of events efficiently is of value (being able to provision a tweaked read model in hours rather than days has significant leverage).
+
+### Versioning read models over time
+
+One aspect to call out is that it's easy to underestimate the frequency at which such re-processing is required and/or is the best approach to apply; both during the initial development phase of a system (where processing may only have happened in pre-production environment so a quick `TRUNCATE TABLE` will cure it all) or for a short time (where a quick online `UPDATE` or `SELECT INTO` can pragmatically address a need). Two pieces of advice arise from this:
+- SQL databases are by far the most commonly used read model stores for good reason - they're malleable (you whack in a join or add an index or two and/or the above mentioned `TRUNCATE TABLE`, `SELECT INTO` tricks will take you a long way), and they provide Good Enough performance and scalability for the vast majority of scenarios (you are not Google).
+- it's important to practice how you play with regard to such situations. Use the opportunity to sharpen you and your teams thinking and communication with regard to how to handle such situations rather than cutting corners eery time. That muscle memory will pay back quicker than you think; always sweeping them under the rug (only every doing hacks) can turn you into an author of one of those sad "what the Internet didn't tell you about Event Sourcing" articles in short order.
+
+In the long run, getting used to dealing with re-traversal scenarios by building handlers to provision fresh adjacent read models is worthwhile. It also a skill that generalises better - a random document store is unlikely to match the full power of a `SELECT INTO`, but ultimately they may be a better overall solution for your read models (`Equinox.CosmosStore` and `Equinox.DynamoStore` also offer powerful `RollingState` modes that can simplify such processing).
+
+In short, it's strongly recommended to at least go through the thought exercise of considering how you'd revise or extend a read model in a way that works when you have a terabyte of data or a billion items in your read model every time you do a 'little tweak' in a SQL read model.  
```

Comment:
With a billion items that little tweak can wreak untold havoc. Consider the case of accidentally locking the entire table to perform a little tweak, or forgetting to create your index `concurrently`. These can have massive consequences.

I think one pattern that's not been called out in this article is to spin up a new model with the tweak and route traffic to the entirely new read model once it's up to date. It provides a very safe experience and an instant rollback (flip the flag back to the old read model). Finally you can truncate and get rid of the old model once you're happy.